src/zenml/zen_server/auth.py

```diff
@@ -38,6 +39,34 @@
 
 logger = get_logger(__name__)
 
+# create a context variable to store the authentication context
+_auth_context: ContextVar[Optional["AuthContext"]] = ContextVar(
+    "auth_context", default=None
+)
+
+
+def get_auth_context() -> Optional["AuthContext"]:
```

Comment:
wow that's a very simple solution to the auth passing problem, nice 👍

Reply:
Props to none other than Mr. @stefannica :)